### ISSUE TITLE
ContactsService.addContact to return id/displayName

### ISSUE THREAD

**Issue by megid2** [lukasgit/flutter_contacts#18]
Not sure if possible with Android/iOS, but I would appreciate if `ContactsService.addContact(contact)` would:
- set `contact.id` and `contact.displayName` as generated upon insertion
- or return a `Future<Contact>` with the newly inserted contact including `id` and `displayName`

My use case is adding a new contact and then immediately switching to the contact update page (or rather going add -> detail -> update). To call `ContactsService.updateContact(contact)`, `contact.id` must be set but I do not know it a priori after adding.

My current workaround is to reload the contact with `ContactsService.getContacts(query = [contact.givenName, contact.middleName, contact.familyName].where((s) => s != null).join (" "))` but this fails e.g. if no `familyName` is set. Loading all contacts and then searching manually would also be possible but time consuming.

In summary, getting the `id`/`displayName` from `addContact` would be a much cleaner solution, and I would much prefer it to my current, somewhat hacky, workaround.

**Comment by z0mbie42** [comment]
Hi,

Any news on this issue?

I encounter the same problem and returning the `identifier` on `addContact` would make it.

**Comment by megid2** [comment]
I have not heard from Lucas. In the meantime, I have cloned the repo and tried getting some progress, but as far as I can see there is not even an instrumented unit test in the android portion and at the moment I am struggling with permissions. The last time I developed Android was in 2010 and it certainly hasnt got any easier.

**Comment by megid2** [comment]
Info: on Android, getting the (raw, not aggregated) contact id can be done in addContact():
```
ContentProviderResult[] addContactResults = contentResolver.applyBatch(ContactsContract.AUTHORITY, ops);
final long rawContactId = ContentUris.parseId(addContactResults[0].uri);
contact.identifier = String.valueOf(rawContactId);
```

**Comment by lukasgit** [comment]
Hi @megid2 and @z0mbie42 if you find something to add to this package, please feel free to submit your PR for review. Thanks!

**Comment by z0mbie42** [comment]
Hi @megid2, great news! But what do you mean by `raw` vs `aggregated`?

**Comment by megid2** [comment]
@z0mbie42 If you add a contact, you do so with an account (e.g. DavX, Google, Exchange etc.), resulting in a "raw" contact. Now suppose you have a raw contact with displayName "Jane Doe" (Google), then add a raw contact "Jane Doe" (Exchange). When reading, you do not lead the two raw contacts, but rather an aggregated/unified contact. So the table looks like this:

```
displayName | raw id | unified id | account
Jane Doe    | 1      | 1          | google
Jane Doe    | 2      | 1          | exchange
```

**Comment by joachimvalente** [comment]
I have a PR that does something like that:
https://github.com/fluttercommunity/flutter_contacts/compare/450a0788ff5faa34fcd584075f315c87d185ffeb..5e699cbefb4ee305bb09b503415e90911467c745
